packages/codegen-ui/lib/generate-form-definition/form-to-component.ts

```diff
@@ -88,7 +88,30 @@ export const fieldComponentMapper = (name: string, formDefinition: FormDefinitio
   return parentGrid(`${name}Grid`, formDefinition.form.layoutStyle, fieldChildren);
 };
 
-export const ctaButtonConfig = (): StudioComponentChild => {
+const resolveCtaLabels = (
+  formDefinition: FormDefinition,
+): { cancelLabel: string; clearLabel: string; submitLabel: string } => {
+  const cancelLabel =
+    formDefinition.buttons?.cancel?.visible && formDefinition.buttons?.cancel?.labelOverride
+      ? formDefinition.buttons?.cancel?.labelOverride
+      : 'Cancel';
+
+  const clearLabel =
+    formDefinition.buttons?.clear?.visible && formDefinition.buttons?.clear?.labelOverride
+      ? formDefinition.buttons?.clear?.labelOverride
+      : 'Clear';
+
+  const submitLabel =
+    formDefinition.buttons?.submit?.visible && formDefinition.buttons?.submit?.labelOverride
+      ? formDefinition.buttons?.submit?.labelOverride
+      : 'Submit';
```

Comment:
Do we need to add localization support?

Reply:
I don't think this something we can add in locally as whatever client app would use these components would also need to have localization support. In the UI console we can do it by passing down the override?

Reply:
Yes, the UI code that will accompany these changes will pass in a default override of localized strings for 'Submit', 'Cancel' and 'Clear'